debugger/src/debug_adapter.rs

```diff
@@ -196,25 +224,321 @@ impl<P: ProtocolAdapter> DebugAdapter<P> {
+        request: Request,
+    ) -> Result<()> {
+        let arguments: WriteMemoryArguments = match self.adapter_type() {
+            DebugAdapterType::CommandLine => match request.arguments.as_ref().unwrap().try_into() {
+                Ok(arguments) => arguments,
+                Err(error) => return self.send_response::<()>(request, Err(error)),
+            },
+            DebugAdapterType::DapClient => match get_arguments(&request) {
+                Ok(arguments) => arguments,
+                Err(error) => return self.send_response::<()>(request, Err(error)),
+            },
         };
-        let data = match get_int_argument(request.arguments.as_ref(), "data", 1) {
-            Ok(data) => data,
-            Err(error) => return self.send_response::<()>(request, Err(error)),
+        let memory_offset = arguments.offset.unwrap_or(0);
+        let address: u32 = if let Ok(address) = parse::<i64>(arguments.memory_reference.as_ref()) {
+            match (address + memory_offset).try_into() {
+                    Ok(modified_address) => modified_address,
+                    Err(error) => return self.send_response::<()>(
+                    request,
+                    Err(DebuggerError::Other(anyhow!(
+                        "Could not convert memory_reference: {} and offset: {:?} into a 32-bit memory address: {:?}",
+                        arguments.memory_reference, arguments.offset, error
+                    ))),
+                ),
+                }
+        } else {
+            return self.send_response::<()>(
+                request,
+                Err(DebuggerError::Other(anyhow!(
+                    "Could not read any data at address {:?}",
+                    arguments.memory_reference
+                ))),
+            );
+        };
+        let data_bytes = match base64::decode(&arguments.data) {
+            Ok(decoded_bytes) => decoded_bytes,
+            Err(error) => {
+                return self.send_response::<()>(
+                    request,
+                    Err(DebuggerError::Other(anyhow!(
+                        "Could not decode base64 data:{:?} :  {:?}",
+                        arguments.data,
+                        error
+                    ))),
+                );
+            }
         };
-
         match core_data
             .target_core
-            .write_word_32(address, data)
+            .write_8(address, &data_bytes)
             .map_err(DebuggerError::ProbeRs)
         {
-            Ok(_) => Ok(()),
+            Ok(_) => {
+                self.send_response(
+                    request,
+                    Ok(Some(WriteMemoryResponseBody {
+                        bytes_written: Some(data_bytes.len() as i64),
+                        offset: None,
+                    })),
+                )?;
+                // TODO: This doesn't trigger the UI to reload the variables effected. Investigate if we can force it in some other way, or if it is a known issue.
```

Comment:
Well, it correctly updates the target memory, so any further processing that depends on that memory on the target will use the new data. Similarly, a new stack trace (e.g. after a step function) will show the newly updated value from the target. 

This binary edit of memory is a new feature in VSCode, so I'd like to see if they fix it before we start building code to workaround their UX glitch :)

I added the TODO so that it will be a reminder to me or future contributors.